### PR TITLE
COG-416: Perry performance: optimize database calls on Login operation

### PR DESCRIPTION
**Pull request by DmitriyRudenko**
### JIRA Issue Link
- [COG-416: Perry performance: optimize database calls on Login operation](https://osi-cwds.atlassian.net/browse/COG-416)

### Tests
- [x] I have included unit tests 
- [ ] I have included integration tests 
- [ ] I have included performance tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

### Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: src/main/java/gov/ca/cwds/data/auth/UserIdDao.java [hunk 1]
```diff
@@ -17,12 +17,20 @@
...
 
+  @Query("SELECT u FROM UserId u "
+      + "LEFT OUTER JOIN FETCH u.staffPerson sp "
+      + "LEFT OUTER JOIN FETCH sp.unitAuthorities ua "
+      + "LEFT OUTER JOIN FETCH ua.assignmentUnit "
+      + "LEFT OUTER JOIN FETCH sp.office "
+      + "LEFT OUTER JOIN FETCH u.privileges p "
+      + "WHERE u.logonId = :logonId "
+      + "AND u.endDate IS NULL "
+      + "AND p.endDate IS NULL "
```
Comment (sorych) [inline src/main/java/gov/ca/cwds/data/auth/UserIdDao.java]: Do privileges and unitAuthorities really need to be checked for "endDate Is null"? 
please doublecheck

Comment (DmitriyRudenko) [reply src/main/java/gov/ca/cwds/data/auth/UserIdDao.java]: fixed

File: src/main/java/gov/ca/cwds/service/CwsUserInfoService.java [hunk 2]
```diff
@@ -92,22 +86,29 @@ private CwsUserInfo getCwsUserInfo(UserId user, String userIdentifier, String st
         .build();
   }
 
+  private boolean isSocialWorker(UserId userId) {
+    return Optional.ofNullable(userId.getPrivileges())
```
Comment (sorych) [inline src/main/java/gov/ca/cwds/service/CwsUserInfoService.java]: should the privilege be checked for endDate is null here?

Comment (DmitriyRudenko) [reply src/main/java/gov/ca/cwds/service/CwsUserInfoService.java]: its done here: UserId.java:71
@Where(clause = "END_DT IS NULL")
private Set<StaffAuthorityPrivilege> privileges;

**Pull request merged**